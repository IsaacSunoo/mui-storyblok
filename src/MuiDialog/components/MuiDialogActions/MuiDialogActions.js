import React, { createElement, lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import DialogActions from '@material-ui/core/DialogActions';
import Storyblok from '../../../utils/Storyblok';

const MuiIconButtonRedirect = lazy(() => import('../../../MuiIconButtonRedirect/MuiIconButtonRedirect'));
const MuiIconButtonHref = lazy(() => import('../../../MuiIconButtonHref/MuiIconButtonHref'));
const MuiIconButtonDownload = lazy(() => import('../../../MuiIconButtonDownload/MuiIconButtonDownload'));
const MuiButtonRedirect = lazy(() => import('../../../MuiButtonRedirect/MuiButtonRedirect'));
const MuiButtonHerf = lazy(() => import('../../../MuiButtonHerf/MuiButtonHerf'));
const MuiButtonDownload = lazy(() => import('../../../MuiButtonDownload/MuiButtonDownload'));

const MuiDialogActions = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiButtonRedirect,
    MuiButtonHerf,
    MuiButtonDownload,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <DialogActions
      className={styles.root}
    >
      <Suspense fallback={<div />}>
        {content.map((item, index) => createElement(
          components[item.component],
          Object.assign(item, { key: index }),
        ))}
      </Suspense>
    </DialogActions>
  );
};

export default MuiDialogActions;

MuiDialogActions.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiButtonRedirect */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiDialogActions.defaultProps = {
  rootClass: [],
};
