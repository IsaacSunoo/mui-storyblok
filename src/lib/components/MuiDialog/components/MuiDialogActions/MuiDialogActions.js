import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import DialogActions from '@material-ui/core/DialogActions';

import Storyblok from 'lib/utils/Storyblok';
import MuiIconButtonRedirect from 'lib/components/MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiIconButtonHref from 'lib/components/MuiIconButtonHref/MuiIconButtonHref';
import MuiIconButtonDownload from 'lib/components/MuiIconButtonDownload/MuiIconButtonDownload';
import MuiButtonRedirect from 'lib/components/MuiButtonRedirect/MuiButtonRedirect';
import MuiButtonHref from 'lib/components/MuiButtonHref/MuiButtonHref';
import MuiButtonDownload from 'lib/components/MuiButtonDownload/MuiButtonDownload';

const MuiDialogActions = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiButtonDownload,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <DialogActions
      className={styles.root}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </DialogActions>
  );
};

export default MuiDialogActions;

MuiDialogActions.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiButtonDownload,
    */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiDialogActions.defaultProps = {
  rootClass: [],
};
