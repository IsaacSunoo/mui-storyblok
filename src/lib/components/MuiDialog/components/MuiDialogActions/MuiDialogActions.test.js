import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiDialogActions from './MuiDialogActions';

function setup() {
  const props = {
    content: [{
      component: 'MuiButtonRedirect',
      redirectRoute: 'page-welcome',
      button: [{
        component: 'MuiButton',
        buttonText: 'buttonText',
      }],
    }],
  };
  const comp = shallow(<MuiDialogActions {...props} />);

  return { comp, props };
}

describe('<MuiDialogActions />', () => {
  it('renders MuiDialogActions', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });


  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create((
      <MemoryRouter>
        <MuiDialogActions {...props} />
      </MemoryRouter>
    ));
    expect(tree).toMatchSnapshot();
  });
});
