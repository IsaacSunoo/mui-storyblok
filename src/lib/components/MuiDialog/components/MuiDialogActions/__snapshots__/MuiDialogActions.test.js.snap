// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiDialogActions /> snapshot 1`] = `
<div
  className="MuiDialogActions-root makeStyles-root-3 MuiDialogActions-spacing"
>
  <div />
</div>
`;
